### PR TITLE
Refactores the Tooltip component into a functional component

### PR DESCRIPTION
**Pull request by grzim**
## Description

Regarding https://github.com/WordPress/gutenberg/issues/22890 "Refactor all React class components to functional components using hooks" this pull request contains a rewritten tooltip

## How has this been tested?

Changes tested visually and with a set of unit tests that were adjusted to work with the functional component.

## Types of changes
Refactoring

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/tooltip/index.js [hunk 1]
```diff
@@ -24,60 +25,69 @@ import Shortcut from '../shortcut';
...
+};
 
-	emitToChild( eventName, event ) {
-		const { children } = this.props;
-		if ( Children.count( children ) !== 1 ) {
-			return;
-		}
+function Tooltip( { children, position, text, shortcut } ) {
+	/**
+	 * Whether a the mouse is currently pressed, used in determining whether
```
Comment (draganescu) [inline packages/components/src/tooltip/index.js:43]: "Whether a the mouse" => "Whether the mouse"

**Review by draganescu — APPROVED** [review]
LGTM, tested and works as expected. @grzim check if the tests pas and maybe fix that nitpick I left about that old comment.

File: packages/components/src/tooltip/index.js (hunk 1 above)
Comment (grzim) [reply packages/components/src/tooltip/index.js]: comment fixed, 
tooltip unit tests are passing

**Comment by grzim** [comment]
resolved in https://github.com/WordPress/gutenberg/pull/27682

**Pull request closed without merging**